tardis/widgets/base.py

```diff
@@ -0,0 +1,546 @@
+from tardis.base import run_tardis
+from tardis.io.atom_data.util import download_atom_data
+from tardis.util.base import atomic_number2element_symbol, species_tuple_to_string
+from tardis.simulation import Simulation
+
+import pandas as pd
+import numpy as np
+import qgrid
+import ipywidgets as ipw
+
+
+class BaseShellInfo():
+    """The simulation information that is used by shell info widget
+    """
+
+    def __init__(self, t_radiative, w, abundance, number_density,
+                 ion_number_density, level_number_density):
+        """Initialize the object with all simulation properties in use
+
+        Parameters
+        ----------
+        t_radiative : array_like
+            Radiative Temperature of each shell of simulation
+        w : array_like
+            Dilution Factor (W) of each shell of simulation model
+        abundance : pandas.DataFrame
+            Fractional abundance of elements where row labels are atomic number
+            and column labels are shell number
+        number_density : pandas.DataFrame
+            Number densities of elements where row labels are atomic number and
+            column labels are shell numbers
+        ion_number_density : pandas.DataFrame
+            Number densities of ions where rows are multi-indexed with (atomic
+            number, ion number) and column labels are shell number
+        level_number_density : pandas.DataFrame
+            Number densities of levels where rows are multi-indexed with (atomic
+            number, ion number, level number) and column labels are shell number
+        """
+        self.t_radiative = t_radiative
+        self.w = w
+        self.abundance = abundance
+        self.number_density = number_density
+        self.ion_number_density = ion_number_density
+        self.level_number_density = level_number_density
+
+    def shells_data(self):
+        """Generates shells data in a form that can be used by a table widget
+
+        Returns
+        -------
+        pandas.DataFrame
+            Dataframe containing Rad. Temp. and W against each shell of
+            simulation model
+        """
+        shells_temp_w = pd.DataFrame({'Rad. Temp.': self.t_radiative,
+                                      'W': self.w})
+        shells_temp_w.index = range(
+            1, len(self.t_radiative)+1)  # Overwrite index
+        shells_temp_w.index.name = 'Shell No.'
+        # Format to string to make qgrid show values in scientific notations
+        return shells_temp_w.applymap(lambda x: '{:.6e}'.format(x))
```

Comment:
there is a way to style dataframes - that might be better

Reply:
Yes I know that, but it doesn't make sense to qgrid widget because they fetch the data from dataframe and show it in the format of slickgrid (JS library they use underneath). I know it is not elegant way but more like a workaround but they have no native way to show numbers in scientific notation - I've also opened an [issue](https://github.com/quantopian/qgrid/issues/320) for it.